SIPS/sip-keeper_synths.md

```diff
@@ -0,0 +1,72 @@
+
+## Abstract
+This SIP will change the iSynths purge and update pricing functions to be public and incentivised. The first person to call each function will be paid SNX as an incentive.
+
+## Motivation
+While the iSynths are extremely useful and somewhat differentiated among DeFi assets, they have significant friction for users as the price bands required to prevent them becoming over or under leveraged mean that they regularly freze and must be reset. This reset process also requires that all holders of the Synth are purged into sUSD. 
+
+The reason the price bands are required is that if the price of the underlying Synth doubles the price of the iSynth will go to 0. As the price of the iSynth tends to 0 the amount of leverage increases. For example if an ETH iSynth was instantated at $200 and ETH went to $399 then the iSynth would go to $1, so you would get $1 of price movement for each $1 spent on iETH versus $1 of price movement for every $400 spent on sETH. This property reverses in the other direction such that as the price of ETH drops you must buy significantly more iETH to get the same level of price movement. 
+
+These properties neccesitate tight price bands and increasing or removing them would add significant risk to the network. The solution proposed in this SIP therefore leaves the bands as is and instead improves the purging and reset process by incentivising anyone to call public functions to manage iSynth freezing. 
+
+## Specification
+<!--The specification should describe the syntax and semantics of any new feature, there are five sections
+1. Overview
+2. Rationale
+3. Technical Specification
+4. Test Cases
+-->
+
+# Overview
+There are three core functions required to reset an iSynth.
+1. FreezeSynth
+2. Purge
+3. ResetPrice
+
+FreeSynth - Currently the oracle is responsible for freezing the iSynth when the price limits are hit, as an aside this is also problematic for our transition to chainlink as the CL oracles do not have the ability to call freeze based on a price limit being hit. Rather than relying on a centralised service to freeze a synth we will instead open this up to be a public function that any address can call. This function will require that the current price of the underlying Synth is outside the defined price bands. If it is the synth will be frozen, if not the call will fail. The fee in SNX for freezing an iSynth will initially be set to 50 SNX. This fee is deliberately set high to ensure there are sufficient incentives for bots to operate on these Synths.
+
+Purge - Currently once a Synth is frozen the pDAO must call purge on all the addresses that hold this iSynth. This will also be made a public function and any address will be able to pass an array of addresses holding these Synths and they will be purged into sUSD. 
```

Comment:
Another idea here is to incentivize individuals to exchange themselves back into sUSD or another synth - either by exposing a version of purge to `msg.sender`. That said, you'd imagine that most holders of an inverse synth would prefer to take no action and have their holdings automatically rebalanced as per the fee reclamation solution discussed below. 

What if instead of incentivizing the purge, we incentivize the reset, and this can only be done once the total supply is below a threshold? This way, the focus will be on being the one to purge and then immediately reset (probably requiring a custom contract to prevent someone frontrunning your reset if you spent the gas to purge). It's riskier to the purger but if they are sophisticated (and motivated enough), they should be able to come up with a strategy to combine the purge and resets smartly.

Reply:
I think the challenge with this is that there need to be multiple rounds of purges and they are too large to be done as atomic transactions. One solution though could be to only incentivise freezeSynth and resetPrice but lock down those incentives to the address that calls purge for say 1 hour. That way the address that freezes has a window of time in which to complete the reset without risking the reward.

Reply:
That could work but how to handle which address did the most purging? As soon as a purge happens it would have to freeze purging as well to only that address for an hour.